src/sagemaker_training/mpi.py

```diff
@@ -177,6 +179,23 @@ def _orted_process():  # pylint: disable=inconsistent-return-statements
         time.sleep(1)
 
 
+def _smddp_command(instance_type):  # type: (str) -> list[str]
+    """When a task is of modelparallel and ddp_dist_backend is auto,
+    we use smddpmprun to set up necessary environment variables if possible.
+    """
+    command = []
+    env = environment.Environment()
+    if env.is_modelparallel_enabled:
+        mp_parameters = json.loads(os.environ.get(params.SM_HP_MP_PARAMETERS, "{}"))
+        ddp_dist_backend = mp_parameters.get("ddp_dist_backend", "auto")
+        if ddp_dist_backend == "auto":
+            if env.is_smddp_smddpmprun_installed:
+                command.extend(["smddpmprun", instance_type, "--allow-bypass"])
+        else:
+            logger.info(f"{ddp_dist_backend} is used as backend for training")
```

Comment:
Do we really need this log? I think we print more detailed information internally.

Reply:
This log wouldn't be accurate if herring was passed but the config doesn't support Herring, right?

Reply:
Our agreement is that we will only have `auto` and `nccl` as options, where `auto` means best effort using the feature. So the branch here corresponds to `nccl`.

Reply:
Yes, but this may change in the future. Since we have internal logic that can potentially modify the backend used, I suggest removing this log to avoid inaccurate logs.

Reply:
@rahul003 Which config are you referring to ? If the passed option isn't supported, aren't we erroring out ?
@dannyel2511 : As SMTT is the main process launcher here and it dumps out the info even before the trianing starts, it would be comprehend the logs from the toolkit.

Reply:
@yongyanrao : Could you please update the wording accordingly?